application/common/models/User.php

```diff
@@ -0,0 +1,190 @@
+
+    public $password;
+
+    public function scenarios(): array
+    {
+        $scenarios = parent::scenarios();
+
+        $scenarios[self::SCENARIO_DEFAULT] = [
+            'employee_id',
+            'first_name',
+            'last_name',
+            'display_name',
+            'username',
+            'email',
+            'active',
+            'locked',
+        ];
+
+        $scenarios[self::SCENARIO_UPDATE_PASSWORD] = ['password'];
+
+        $scenarios[self::SCENARIO_AUTHENTICATE] = ['username', 'password'];
+
+        return $scenarios;
+    }
+
+    public function rules(): array
+    {
+        return ArrayHelper::merge([
+            [
+                'active', 'default', 'value' => 'yes',
+            ],
+            [
+                'locked', 'default', 'value' => 'no',
+            ],
+            [
+                ['active', 'locked'], 'in', 'range' => ['yes', 'no'],
+            ],
+            [
+                'email', 'email',
+            ],
+            [
+                'password', 'required',
+                'on' => [self::SCENARIO_UPDATE_PASSWORD, self::SCENARIO_AUTHENTICATE],
+            ],
+            [
+                'password', 'string',
+            ],
+            [
+                'password',
+                $this->validatePassword(),
+                'on' => self::SCENARIO_AUTHENTICATE,
+            ],
+            [
+                ['last_synced_utc', 'last_changed_utc'],
+                'default', 'value' => Utils::now(),
+            ],
+            [
+                ['employee_id', 'first_name', 'last_name', 'display_name', 'username', 'email'],
+                'string',
+            ],
+        ], parent::rules());
+    }
+
+    private function validatePassword(): \Closure
+    {
+        return function ($attributeName) {
+            if (! password_verify($this->$attributeName, $this->password_hash)) {
+                $this->addError($attributeName, 'Incorrect password.');
+            }
+        };
+    }
+
+    public function behaviors(): array
+    {
+        return [
+            'changeTracker' => [
+                'class' => AttributeBehavior::className(),
+                'attributes' => [
+                    ActiveRecord::EVENT_BEFORE_INSERT => 'last_changed_utc',
+                    ActiveRecord::EVENT_BEFORE_UPDATE => 'last_changed_utc',
+                ],
+                'value' => Utils::now(),
+                'skipUpdateOnClean' => true, // only update the column if the model is dirty
+            ],
+            'updateTracker' => [
+                'class' => AttributeBehavior::className(),
+                'attributes' => [
+                    ActiveRecord::EVENT_BEFORE_INSERT => 'last_synced_utc',
+                    ActiveRecord::EVENT_BEFORE_UPDATE => 'last_synced_utc',
+                ],
+                'value' => Utils::now(),
+                'skipUpdateOnClean' => false, // always update the column regardless of dirtiness
```

Comment:
Ahhh, that's an excellent consideration.  I have been viewing the sync more broadly representing the idea of the fact that the information is accurate as of some point, not solely a field that should only be affected by our idsync module...if that's the case we will need to ensure we can distinguish POST's based on consumer so we know not to update the `last_synced` column when a request has come in to update user info, e.g., user mgmt...I think that's going to require some additional spec.  At this point that might be as simple as assuming that _only_ POST's to `/user` affect that `last_synced_utc` column under the assumption the only consumer doing that is the id-sync, sound right?